pkg/api/pod/warnings.go

```diff
@@ -274,6 +274,8 @@ func warningsForPodSpecAndMeta(fieldPath *field.Path, podSpec *api.PodSpec, meta
 
 	// Accumulate ports across all containers
 	allPorts := map[string][]portBlock{}
+	// Accumulate port names of all containers
+	allPortsNames := map[string]*field.Path{}
 	pods.VisitContainersWithPath(podSpec, fieldPath.Child("spec"), func(c *api.Container, fldPath *field.Path) bool {
```

Comment:
hmm... this iterates over all containers: initContiners, ephemeralContainers, and containers

what happens if a named port is in a container in all three lists? is the routing still ambiguous, or will the one in `spec.containers` still be used? the places I see us resolving ports look only at the `spec.containers` list

what if the one in initContainers is in a sidecar container that keeps running?

Reply:
I think we should still warn if ANY two containers use the same port name (which I think this will do, but should add tests to cover)

Seperately, we should consider if services and probes should hunt for ports by name inside initContainers (I can't see why hunting inside ephemeral would make sense).

Reply:
I think only normal containers and sidecar containers should be checked. 
In addition，ports are not allowed for ephemeral containers.  @thockin @liggitt